### ISSUE TITLE
NDB: fdb multicast dump missing records

### ISSUE THREAD

**Issue by pkulev** [svinota/pyroute2#1091]
pyroute2 version: 0.6.13 and 0.7.7

Steps to reproduce:

I'm using a little homebrew cache for multicast records:
```python
In [17]: constants.MCAST_LLADDR
Out[17]: '00:00:00:00:00:00'

In [18]: vtep.fdb.refresh??
Signature: vtep.fdb.refresh() -> 'None'
Docstring: Clear the cache and fill data.
Source:
    def refresh(self) -> None:
        self.clear()

        for fdb in self.root.ip.fdb("dump"):
            if (
                fdb.get_attr("NDA_DST") is not None
                and fdb.get_attr("NDA_LLADDR") == constants.MCAST_LLADDR
            ):
                entry = FDB(
                    fdb["ifindex"],
                    fdb.get_attr("NDA_LLADDR"),
                    fdb.get_attr("NDA_DST"),
                )
                self.fdb.setdefault(entry.index, {})[entry.dst] = entry
Type:      method
```

If we refresh it and show the internals, there will be two records.

```python
In [19]: vtep.fdb.refresh()

In [20]: vtep.fdb.fdb
Out[20]:
{32: {'10.1.0.107': FDB(index=32, lladdr='00:00:00:00:00:00', dst='10.1.0.107'),
  '10.1.0.11': FDB(index=32, lladdr='00:00:00:00:00:00', dst='10.1.0.11')}}
```

If we will do similar filtering using NDB: we will get only one record:
```python
In [21]: fdb_dump = ndb.fdb.dump()

In [22]: fdb_dump.select_records(lladdr=constants.MCAST_LLADDR, dst=lambda dst: dst is not None)

In [23]: fdb_dump.select_fields("lladdr", "dst", "ifindex")

In [24]: fdb_dump
Out[24]: ('00:00:00:00:00:00', '10.1.0.11', 32)
```

bridge fdb dump shows also two records:
```
~# bridge fdb show | grep 00:00:00:00:00:00
00:00:00:00:00:00 dev vxln7 dst 10.1.0.107 self extern_learn permanent
00:00:00:00:00:00 dev vxln7 dst 10.1.0.11 self extern_learn permanent
```

It looks like overriding with lastly added record.
If I manually add a new record via `bridge fdb append` NDB notices this and shows only the last record in the same dump.

How can I help you in investigating that mismatch? Maybe there is workaround?

**Comment by svinota** [comment]
Thanks, investigating

**Comment by pkulev** [comment]
Fixed in #1158
